File: clients/python-component/examples/produce.py
#!/usr/bin/env python

from Broker.Messages import Message, Publish
from Broker.Transport import TCP, UDP
from Broker.Codecs import Codec #auto codec selection (thrift or protobuf if thrift isn't installed)
from Broker.Clients import Minimal

server='localhost'
destination = '/python/tests'
destination_type = 'QUEUE'
N=10000

broker = Minimal(codec=Codec(), transport=TCP(host=server))

for n in xrange(N):
    message = Message(payload='Message number %d' % n)
    publish = Publish(destination=destination, destination_type=destination_type, message=message)
    broker.send(publish)

File: clients/perl-component/examples/produce.pl
use lib ('../lib');

use SAPO::Broker::Clients::Simple;

use strict;
use warnings;

my $broker  = SAPO::Broker::Clients::Simple->new(host=>'broker.m3.bk.sapo.pt',port=>2222);
my %options = (
    'destination_type' => 'QUEUE',
    'destination'      => '/tests/perl',
);

my $N      = $ARGV[0] || 100;
my $prefix = $ARGV[1] || ( "time=" . time );

for my $n ( 1 .. $N ) {
    $broker->publish( %options, 'payload' => "$prefix\tpayload $n" );
}
